File: lab03/faculty_mfiles/spect.m
function [Pw Fq] = spect (x, Fs)
L = length(x);             % Length of signal
P = abs(fft(x)/L);
Pw = P(1:floor(L/2)+1);
Pw(2:end-1) = 2*Pw(2:end-1);
Fq = Fs*(0:L/2-1)/L;
if length(Pw)> length(Fq) 
    Pw = Pw(1:length(Fq));
else
    Fq = Fq(1:length(Pw));
end

File: lab04/spect.py
from scipy.fft import fft
from numpy import arange
from math import floor


def spect(signal, sf):
    l = len(signal)
    p = abs(fft(signal) / l)
    pw = p[1:floor(l / 2) + 1]
    pw[2: - 1] = 2 * pw[2: - 1]
    fq = sf * arange(l/2-1) / l
    if len(pw) > len(fq):
        pw = pw[:len(fq)]
    else:
        fq = fq[:len(pw)]
    return pw, fq

